packages/teams-js/src/public/visualMedia.ts

```diff
@@ -0,0 +1,274 @@
+import { sendAndHandleSdkError } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { errorNotSupportedOnPlatform, FrameContexts } from './constants';
+import { DevicePermission } from './interfaces';
+import { runtime } from './runtime';
+
+/**
+ * Interact with image and video. It lets the app developer ask the user to get images or video from their camera / camera roll / file system.
+ * @beta
+ */
+export namespace visualMedia {
+  /**
+   * @hidden
+   * Hide from docs
+   * --------
+   * All properties common to Image and Video Props
+   * @beta
+   */
+  interface VisualMediaProps {
+    /**
+     * @hidden
+     * Lets the developer specify the media source
+     */
+    source: Source;
+
+    /**
+     * @hidden
+     * Optional; Specify in which mode the camera will be opened.
+     * Default value is Photo
+     */
+    startMode?: CameraStartMode;
+
+    /**
+     * @hidden
+     * Optional; indicate if user is allowed to move between front and back camera
+     * Default value is true
+     */
+    cameraSwitcher?: boolean;
+    /**
+     * max limit of media allowed to be selected in one go, current max limit is 10 set by lens-sdk.
+     */
+    visualMediaCount: number;
+  }
```

Comment:
The current design of this interface allows for the developer to specify undefined/confusing combinations. For example, the `startMode` property only appears to apply to the camera source, but the current design allows the developer to specify a `startMode` even if the `source` is `Gallery`.  I think you can slightly reshape the interface to make it so that the developer can only specify property values that align with the source type for which they make sense, e.g.,

```
interface GalleryProps {
    source: Source.Gallery;
}

interface CameraProps {
    source: Source.Camera; 
    startMode?: CameraStartMode;
    cameraSwitcher?: boolean;
}

interface VisualMediaProps {
    sourceProps: GalleryProps | CameraProps;
}

const myProps: ImageProperties = { sourceProps: { source: Source.Gallery } };
const myProps2: ImageProperties = { sourceProps: { source: Source.Camera } };
const myProps3: ImageProperties = { sourceProps: { source: Source.Camera, startMode: CameraStartMode.Photo } };
const myProps3: ImageProperties = { sourceProps: { source: Source.Gallery, startMode: CameraStartMode.Photo } }; // Correctly won't compile
```

Reply:
The reason of not separating into CameraProps and GalleryProps is: In the captureImages() flow, in the host app, users can also select images from the gallery while in startMode. The result will contain a combination of images from different sources in a list. 
This case only occurs in the captureImages() flow, not in uploadImages(), as users can only access the gallery in the uploadImages() flow.

Reply:
Make sense! If the image comes from the gallery, it is confusing to have startMode and cameraSwitcher.